### DISCUSSION TITLE
Flows: wow! But also: how?

### DISCUSSION THREAD

**Discussion by Autr** [directus/directus#16024] [Archive]
Hello, just had a quick play around with Flows - it looks really intriguing and promising - yet I'm unable to get past even the first hurdle (ie. how do Options and Payloads interact with JSON object keys?) 😅

I have 10,000 Mixcloud entries that have been scraped into a Directus database for a community radio station (these entries are now called _Recordings_). And I'm curious if I could give an option to the admins to do this in reverse - ie. press a button to upload future Recordings to Mixcloud.

The first thing I try to do is grab the Recording entry data into the flow, like this:

- **Manual** block is enabled for Items in Recordings
- When I click this, I see a **Payload** is returned with `$trigger.body.keys` (?)
- **Read Data** block then tries to use this, ie. IDs field is set to `$trigger.body.keys` or `$trigger.body.keys[0]`

... but nothing like this seems to come through.

Based on the assumption that Options means _incoming-payload-to-the-block_, and Payload means _outgoing-response-from-the-block_, in terms of RESTful behaviour.

So are there any examples or more docs I could look at?

The eventual Flow (which would be amazing, if it can be done entirely by a GUI), would be:

- Recordings get manual Trigger button...
- A Mixcloud ID is constructed from my Recording entry, ie. `https://api.mixcloud/{{entry.name}}/{{entry.title}}-{{entry.date}}`
- GET is done...
- If 500, makes alert: "Recording is already uploaded", sets a boolean field `mixcloud_exists=true`
- if 404, constructs a POST request to Mixcloud `https://api.mixcloud/{{entry.name}}/{{entry.title}}-{{entry.date}}` + params, MP3 etc
- if 500, makes alert: "Recording was uploaded", sets boolean `mixcloud_exists=true`
- If !500, makes alert: "Could not upload {{err.message}}", sets boolean `mixcloud_exists=false`

I would also be happy to write some documentation for this feature if I can work it out.

And finally thank you for your hard work! Directus continues to amaze and delight me.

**Comment by erondpowell** [comment]
Howdy, I'm Eron, the technical writer. Yeah, good questions! Cool flow concept 🤠 

> So are there any examples or more docs I could look at?

We're on the last stretches of a big overhaul to Flows docs. After that I'm starting on a flows "cookbook" to help people implement flows for a broad range of use-cases. If you're interested, I'd love to set a time to do a user interview: hear a little about your background, see what you're liking about flows, talk me through your flow and any pain-points, and get an all-around feel for how its working out for you. 

> ... but nothing like this seems to come through.

Can we get some more details on your flow config?
How did you setup the Manual trigger and read operations?
When you click the trigger button, are you selecting items?
Can you show us the Logger panel details as well?

**Comment by emilyf** [comment]
hey there! I wasn't sure where to put this but I've been struggling w/ flows too and after lots of digging and reading i thought the least i could do is put together a quick video that helps with a lot of what you're touching on @autr until the docs are finished. I was looking for a place to attach this and this thread seems pretty good! I hope this helps, given how amazing the Directus team and community is I know there's a lot of us itching to get more info on how flows works! This is just the tip of the iceberg but I hope it can help reduce confusion and team time needed to keep answering all of us in the interim. 

https://www.youtube.com/watch?v=BxrXQST-wzw

**Comment by Autr** [comment]
Wow, thanks @emilyf ! That is a great a help! 🦾
